index.ts

```diff
@@ -1,31 +1,38 @@
 import { multiUrlGenerator } from './lib/multiUrlGenerator';
 import fetch from 'node-fetch';
+import console from 'console';
+import yargs from 'yargs';
+import { generalTransformer } from './config/parsers/ulr.parser';
+const PATH = './config/url_components/';
+const MODE = yargs.argv._[0];
 
-// Test fetching Wiki articles
-const requestUrls = multiUrlGenerator(
-  'https://$lang.wikipedia.org/wiki/$article',
-  [
-    { $lang: 'en', $article: 'Plato' },
-    { $lang: 'es', $article: 'Aristóteles' },
-  ]
-);
+async function getUrls(file: string) {
+  const requestUrls = multiUrlGenerator(
+    await import(PATH + file)
+      .then((jsonFile) => jsonFile)
+      .then((data) => data.linkto),
 
-async function main() {
-  for (const url of requestUrls) {
+    await import(PATH + file)
+      .then((jsonFile) => jsonFile)
+      .then((data) => data.components)
+  );
+
+  return new Promise<string[]>((resolve, reject) => {
+    resolve(requestUrls);
+    reject('no files found');
+  });
+}
+
+async function main(mode: string) {
+  for (const url of await getUrls(`${mode}.urls.json`).then((x) => x)) {
     try {
-      const response = await fetch(url);
-      console.log(
-        `Called url: ${url} - with response status:`,
-        response.status
-      );
+      const response = await fetch('https://' + url);
```

Comment:
"https://" should actually be a part of the base template in the config file. we should not assume the https protocol, so let's just call fetch(url) here